nifi-nar-bundles/nifi-aws-bundle/nifi-aws-parameter-value-providers/src/main/java/org/apache/nifi/stateless/parameter/aws/SecretsManagerParameterValueProvider.java

```diff
@@ -0,0 +1,172 @@
+    public String getParameterValue(final String contextName, final String parameterName) {
+        final String secretName = getSecretName(contextName, parameterName);
+        final GetSecretValueRequest getSecretValueRequest = new GetSecretValueRequest()
+                .withSecretId(secretName);
+        final GetSecretValueResult getSecretValueResult = secretsManager.getSecretValue(getSecretValueRequest);
+
+        if (getSecretValueResult == null) {
+            throw new IllegalArgumentException(String.format("Secret [%s] not found", secretName));
+        }
+        if (getSecretValueResult.getSecretString() != null) {
+            return getSecretValueResult.getSecretString();
+        } else {
+            throw new IllegalStateException(String.format("Secret Name [%s] string value not found", secretName));
+        }
+    }
+
+    private void cacheSupportedParameterNames() {
+        supportedParameterNames.clear();
+        final ListSecretsResult listSecretsResult = secretsManager.listSecrets(new ListSecretsRequest());
+        if (listSecretsResult != null) {
+            for (final SecretListEntry entry : listSecretsResult.getSecretList()) {
+                supportedParameterNames.add(entry.getName());
+            }
+        }
+    }
+
+    private Properties loadProperties(final String propertiesFilename) throws IOException {
+        final Properties properties = new Properties();
+
+        try (final InputStream in = new FileInputStream(Paths.get(propertiesFilename).toFile())) {
+            properties.load(in);
+            return properties;
+        }
+    }
+
+    AWSSecretsManager configureClient(final String awsCredentialsFilename) throws IOException {
+        if (awsCredentialsFilename == null) {
+            return getDefaultClient();
+        }
+        final Properties properties = loadProperties(awsCredentialsFilename);
+        final String accessKey = properties.getProperty(ACCESS_KEY_PROPS_NAME);
+        final String secretKey = properties.getProperty(SECRET_KEY_PROPS_NAME);
+        final String region = properties.getProperty(REGION_KEY_PROPS_NAME);
+
+        if (isNotBlank(accessKey) && isNotBlank(secretKey) && isNotBlank(region)) {
+            return AWSSecretsManagerClientBuilder.standard()
+                    .withRegion(region)
+                    .withCredentials(new AWSStaticCredentialsProvider(new BasicAWSCredentials(accessKey, secretKey)))
+                    .build();
+        } else {
+            return getDefaultClient();
+        }
+    }
+
+    private AWSSecretsManager getDefaultClient() {
+        return AWSSecretsManagerClientBuilder.standard()
+                .withCredentials(DefaultAWSCredentialsProviderChain.getInstance())
+                .build();
+    }
+
+    private static String getSecretName(final String contextName, final String parameterName) {
+        return contextName == null ? parameterName : String.format(QUALIFIED_SECRET_FORMAT, contextName, parameterName);
```

Comment:
@gresockj sorry for the confusion. The context name will never be ignored. Rather, what I was suggesting is that it should be ignored if there's no secret name that uses it. For example, if the Context Name is MyContext and the parameter name is MyParam, then `isParameterDefined` should return true if there's a parameter named `MyContext/MyParam` OR if there's a parameter named `MyParam`. Similarly, when `getParameterValue` is called, if there's a secret named `MyContext/MyParam`, then its value should be returned. Otherwise, return the value for the `MyParam` secret. Does that make sense?

Reply:
Ah, I see -- it's fallback logic, rather than simply handling a null Context.

Reply:
Yes, exactly. Because 90% of the time a stateless flow will only use a single Parameter Context. So we don't want to require that they use the name of the parameter context each time. But, if they have a secret named "Password" or something like that, we do want to support disambiguating that.